isis/tests/TestCsmPlugin.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+
+#include "TestCsmPlugin.h"
+#include "TestCsmModel.h"
+
+// Static Instance of itself
+const TestCsmPlugin TestCsmPlugin::m_registeredPlugin;
+
+TestCsmPlugin::TestCsmPlugin() {
+  // deliberately blank
+}
+
+TestCsmPlugin::~TestCsmPlugin() {}
+
+std::string TestCsmPlugin::getPluginName() const {
+  return "TestCsmPlugin";
+}
+
+std::string TestCsmPlugin::getManufacturer() const {
+  return "TestCsmPluginCreator";
+}
+
+std::string TestCsmPlugin::getReleaseDate() const {
+  return "TestDate";
+}
+
+csm::Version TestCsmPlugin::getCsmVersion() const {
+  return csm::Version(3,0,3);
+}
+
+size_t TestCsmPlugin::getNumModels() const {
+  return 2;
+}
+
+std::string TestCsmPlugin::getModelName(size_t modelIndex) const {
+  return "TestModelName";
+}
+
+std::string TestCsmPlugin::getModelFamily(size_t modelIndex) const {
+  return "TestModelFamily";
+}
+
+csm::Version TestCsmPlugin::getModelVersion(const std::string& modelName) const { 
+  return csm::Version(3,0,3);
+}
+
+bool TestCsmPlugin::canModelBeConstructedFromState(const std::string& modelName,
+                                    const std::string& modelState,
+                                    csm::WarningList* warnings) const {
+  try {
+    csm::Model *model = constructModelFromState(modelState, warnings);
```

Comment:
This is a memory leak, model is never deleted.

Reply:
You're right, thank you.

Reply:
Is this a memory leak in usgscsm, too?

Reply:
It would be, we should check that.

Reply:
As it turns out, if you delete the plugin pointer, but don't remove it from the plugin list, it causes a segfault

Reply:
This isn't a plugin pointer though, this is a dangling model pointer?